apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1299 @@
+	}
+
+	/* Set send and receive timeout values */
+	tv.tv_sec = WEBCLIENT_CONF_TIMEOUT_MSEC / 1000;
+	tv.tv_usec = (WEBCLIENT_CONF_TIMEOUT_MSEC % 1000) * 1000;
+	PRNT("recv timeout is set to (%d.%d)sec", tv.tv_sec, tv.tv_usec);
+	ret = setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO,
+				   (struct timeval *)&tv, sizeof(struct timeval));
+	if (ret < 0) {
+		PRNT("ERROR: setsockopt failed %d", ret);
+		return WGET_SOCKET_CONNECT_ERR;
+	}
+
+	/* Get the server address from the host name */
+	server.sin_family = AF_INET;
+	server.sin_port = htons(ws->port);
+	ret = ws_test_wget_gethostip(ws->hostname, &server.sin_addr.s_addr);
+	if (ret < 0) {
+		/* Could not resolve host (or malformed IP address) */
+		PRNT("ERROR: Failed to resolve hostname %d", ret);
+		close(sockfd);
+		return WGET_SOCKET_CONNECT_ERR;
+	}
+
+	/* Connect to server.  First we have to set some fields in the
+	 * 'server' address structure.  The system will assign me an arbitrary
+	 * local port that is not in use.
+	 */
+	ret = connect(sockfd, (struct sockaddr *)&server, sizeof(struct sockaddr_in));
+	if (ret < 0) {
+		PRNT("ERROR: connect failed: %d errno: %d", ret, errno);
+		close(sockfd);
+		return WGET_SOCKET_CONNECT_ERR;
+	}
+	FUNC_EX;
+	return sockfd;
+}
+
+#ifdef CONFIG_NET_SECURITY_TLS
+static int ws_test_wget_base(void *arg)
+{
+	FUNC_EN;
+	int sockfd = -1;
+	int ret;
+	int buf_len, sndlen, len;
+	int encoding = CONTENT_LENGTH;
+	int state = HTTP_REQUEST_HEADER;
+	struct http_message_len_t mlen = {0,};
+	struct ws_test_wget_s ws;
+	struct http_client_request_t *param = (struct http_client_request_t *)arg;
+	struct http_client_response_t response = {0, };
+	int read_finish = false;
+	int send_len;
+
+	struct http_client_tls_t *client_tls = (struct http_client_tls_t *)malloc(sizeof(
+			struct http_client_tls_t));
+	int handshake_retry = WEBCLIENT_CONF_HANDSHAKE_RETRY;
+
+	if (client_tls == NULL) {
+		PRNT("memory not allocated");
+		return -1;
+	}
+
+	/* Initialize the state structure */
+	memset(&ws, 0, sizeof(struct ws_test_wget_s));
+	ws.buffer = param->buffer;
+	ws.buflen = param->buflen;
+
+	/* Parse the hostname (with optional port number) and filename from the URL */
+	ret = netlib_parsehttpurl(param->url, &ws.port, ws.hostname, WS_TEST_MAXHOSTNAME, ws.filename, WS_TEST_MAXFILENAME);
+	if (ret != 0) {
+		PRNT("ERROR: Malformed HTTP URL: %s", param->url);
+		free(client_tls);
+		return WGET_ERR;
+	}
+
+	PRNT("hostname='%s' filename='%s'", ws.hostname, ws.filename);
+
+	if (param->tls && ws_test_webclient_tls_init(client_tls, &param->ssl_config)) {
+		PRNT("Fail to client tls init");
+		goto errout_before_tlsinit;
+	}
+
+	/* Re-initialize portions of the state structure that could have
+	 * been left from the previous time through the loop and should not
+	 * persist with the new connection.
+	 */
+	ws.httpstatus = HTTPSTATUS_NONE;
+	ws.offset = 0;
+	ws.datend = 0;
+	ws.ndx = 0;
+
+	if ((sndlen = ws_test_wget_msg_construct(ws.buffer, param, &ws)) <= 0) {
```

Comment:
Is it ok if `ws_test_wget_msg_construct` returns `0`?

Reply:
Made required changes